plugins/versionlock.py

```diff
@@ -155,13 +156,21 @@ def run(self):
                 cmd = self.opts.subcommand
 
         if cmd == 'add':
-            _write_locklist(self.base, self.opts.package, self.opts.raw, True,
-                            "\n# Added locks on %s\n" % time.ctime(),
-                            ADDING_SPEC, '')
+            entry = _search_locklist(self.opts.package)
+            if entry == '':
+                _write_locklist(self.base, self.opts.package, self.opts.raw, True,
+                                "\n# Added lock on %s\n" % time.ctime(),
+                                ADDING_SPEC, '')
+            else:
+                logger.info("%s %s", EXISTING_SPEC, entry)
```

Comment:
Sorry, yes, you're right, it really should be an error if the action requested conflicts with what's already recorded. I've made some changes to return an error in the case of conflict, and to be more specific in general. (That means the output messages will now differ from the sample test cases you'd already set.) There's a distinction between the solution being "exactly the same pattern" and the user entering a different argument that ends up resolving into the same pattern. So that's why I've used the phrase "recorded in equivalent form" in the non-error condition.  (And then, of course, the pattern recorded is displayed in the message.)

I assumed you want to use dnf.exceptions.Error, since logger.error doesn't set ERRNO on exit. (And I tried to match the Pythonic idiom used elsewhere in the code base, which values terseness.)

Reply:
Yes, raising dnf.exceptions.Error is exactly what I meant.